pkg/awsManager/vpcVpnManager.go

```diff
@@ -0,0 +1,629 @@
+	_ = DeleteVpnConnections(client)
+}
+
+func DeleteVpcInstances(client clientpkg.Client, vpcToBeDeleted []*string) error {
+
+	errFlag := false
+	for _, vpcId := range vpcToBeDeleted {
+
+		//need to clean out the dependencies
+		// EC2 + S3 are already cleaned out by now
+		// delete vpc endpoints
+		err := DeleteVpcEndpoint(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// clear out all ELB
+		err = DeleteELB(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// clear out all network load balancer
+		err = DeleteNetworkLoadBalancer(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// delete NAT gateway
+		err = DeleteNatgateway(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// detach and delete network interface
+		err = DetachAndDeleteNetworkInterface(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//detach any internet gateway , and delete it
+		err = DeleteGateway(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//detach VPN gateway
+		err = DetachVpnGateway(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//cleaning network ACL
+		err = DeleteNetworkAcl(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//now cleaning route tables
+		err = DeleteRouteTables(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		// now cleaning subnets associated with that vpc id
+		err = DeleteSubnetsForVPC(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+		//now cleaning security groups
+		err = DeleteSecurityGroups(client, vpcId)
+		if err != nil {
+			errFlag = true
+		}
+
+		_, err = client.DeleteVpc(&ec2.DeleteVpcInput{VpcId: vpcId})
+		if err != nil {
+			errFlag = true
+			fmt.Println("ERROR deleting VPC :", *vpcId)
+			fmt.Println(err)
+		}
+	}
+
+	if errFlag == true {
+		return errors.New("ERROR")
+	} else {
+		fmt.Println("VPC's deleted successfully for this region  ")
+		return nil
+	}
+
+}
+func DeleteELB(client clientpkg.Client, vpcID *string) error {
+
+	errFlag := false
+	var marker *string
+	for {
+		elbList, err := client.DescribeLoadBalancers(&elb.DescribeLoadBalancersInput{Marker: marker})
+		if err != nil {
+			fmt.Println("ERROR: cant list ELB")
+			fmt.Println(err)
+			errFlag = true
+		}
+
+		for _, elasticLoadBalancer := range elbList.LoadBalancerDescriptions {
+			if *elasticLoadBalancer.VPCId == *vpcID {
+				_, err := client.DeleteLoadBalancer(&elb.DeleteLoadBalancerInput{LoadBalancerName: elasticLoadBalancer.LoadBalancerName})
+				if err != nil {
+					fmt.Println("ERROR: cant delete ELB", *elasticLoadBalancer.LoadBalancerName)
+					fmt.Println(err)
+					errFlag = true
+				}
+			}
+		}
+
+		if elbList.NextMarker != nil {
+			marker = elbList.NextMarker
+		} else {
+			break
+		}
+
+	}
+
+	if errFlag == true {
+		fmt.Println("ERROR : Some ELB not removed. Refer to logs above.")
+		return errors.New("ERROR")
```

Comment:
Can you make these errors more descriptive?

Reply:
done